src/MySqlConnector/MySqlClient/MySqlDataReader.cs

```diff
@@ -287,6 +287,24 @@ private void DoClose()
 				var connection = Command.Connection;
 				if (!connection.BufferResultSets)
 					connection.Session.FinishQuerying();
+
+				if (Command.IsCanceled)
+				{
+					// KILL QUERY will kill a subsequent query if the command it was intended to cancel has already completed.
+					// In order to handle this case, we issue a dummy query to catch the QueryInterrupted exception.
+					// See https://bugs.mysql.com/bug.php?id=45679
+					var killClearCommand = new MySqlCommand("SELECT * FROM fake_table LIMIT 0;", connection);
```

Comment:
`DO SLEEP(0)` is even better (for some reason I was assuming it had to be a non-zero wait, but `SLEEP(0)` seems to work fine).